File: src/algorithms/arrays/two-number-sum/solution1.go
package main

// O(n^2) time | O(1) space
func TwoNumberSum(array []int, target int) []int {
	for i := 0; i < len(array)-1; i++ {
		firstNum := array[i]
		for j := i+1; j < len(array); j++ {
			secondNum := array[j]
			if firstNum+secondNum == target {
				return []int{firstNum, secondNum}
			}
		}
	}
	return []int{}
}

File: src/algorithms/arrays/two-number-sum/solution1.ts
// O(n^2) Time | O(1) Space
export function twoNumberSum(array: number[], targetSum: number) {
    for (let i = 0; i < array.length - 1; i++) {
        const firstNum = array[i];
        for (let j = i + 1; j < array.length; j++) {
            const secondNum = array[j];
            if (firstNum + secondNum === targetSum) {
                return [firstNum, secondNum];
            }
        }
    }
    return [];
}
